### PR TITLE
fix(ci): fix flaky test

### PR DESCRIPTION
**Pull request by sakoush**
**What this PR does / why we need it**:

This PR fixes a flaky test.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: scheduler/pkg/agent/server_test.go [hunk 1]
```diff
@@ -1062,9 +1066,9 @@ func TestSubscribe(t *testing.T) {
 				}(idx, s)
 			}
 
-			time.Sleep(10 * time.Second)
-
-			g.Expect(len(server.agents)).To(Equal(test.expectedAgentsCountAfterClose))
+			for len(server.agents) != test.expectedAgentsCountAfterClose {
```
Comment (lc525) [inline scheduler/pkg/agent/server_test.go:1069]: this will just sleep forever if the test fails, right?

Comment (sakoush) [reply scheduler/pkg/agent/server_test.go:1069]: yes thats true, I can also cap it to a max count?

**Review by lc525 — APPROVED** [review]
lgtm after adding max count in the for loops

File: scheduler/pkg/agent/server_test.go (hunk 1 above)
Comment (sakoush) [reply scheduler/pkg/agent/server_test.go]: done in https://github.com/SeldonIO/seldon-core/pull/6022/commits/dfb347230c479062d310d00f1edce4e1df41b6e4

**Pull request merged**